spec/unit/git/rugged/credentials_spec.rb

```diff
@@ -37,8 +37,20 @@
   describe "generating ssh key credentials" do
     after(:each) { R10K::Git.settings.reset! }
 
+    it "prefers a key from CLI" do
+      allow(File).to receive(:readable?).with("/etc/puppetlabs/r10k/ssh/clikey-id_rsa").and_return true
```

Comment:
Should we also test the case where the file is not readable? I know we already test that below, but that specifically tests the `private_key` setting, rather than `private_key_from_cli` (or the per-repo private key, for that matter). I know it all turns out the same, since we check for readability after choosing which key to use anyway, but that feels like an implementation detail?

Reply:
Sure I can.